subcommand/acl-controller/command_test.go

```diff
@@ -223,7 +233,24 @@ func checkConsulResources(t *testing.T, consulClient *api.Client, expPolicyRules
 		require.Equal(t, partitions[1].Name, testPartitionName)
 	}
 
-	// Check the client policy is created.
+	// Check if the node got registered properly
+	partitionToCheck := ""
+	if partitionsEnabled {
+		partitionToCheck = testPartitionName
+	}
+	nodes, _, err := consulClient.Catalog().Nodes(&api.QueryOptions{Partition: partitionToCheck})
```

Comment:
minor/nit: You could use an [API filter here](https://developer.hashicorp.com/consul/api-docs/catalog#filtering) to filter by Node name:

```suggestion
	nodes, _, err := consulClient.Catalog().Nodes(&api.QueryOptions{
		Partition: partitionToCheck,
		Filter:    fmt.Sprintf("Node == %q", clusterARN),
	})
```

And then you can just check `require.Len(t, nodes, 1)` instead of the following loop (and if you want to be extra sure, also `require.Equal(t, nodes[0].Node, clusterARN)`)